compute_endpoint/globus_compute_endpoint/endpoint/interchange.py

```diff
@@ -360,11 +367,39 @@ def process_pending_tasks():
                     if self.time_to_quit:
                         self.stop()
                         continue  # nominally == break; but let event do it
+
                     try:
-                        incoming_task = self.pending_task_queue.get(timeout=1)
-                        task_msg = unpack(incoming_task)
-                        if not isinstance(task_msg, Task):
-                            raise InvalidMessageError()
+                        prop_headers, body = self.pending_task_queue.get(timeout=1)
+
+                        fid = prop_headers.get("function_id")
+                        tid = prop_headers.get("task_id")
+                        if not fid or not tid:
+                            # Shouldn't occur live but tests may not set headers
+                            log.exception("Task message missing headers")
```

Comment:
- What exception has occurred at this point?
- If either are `None`, shouldn't this `continue`?

Reply:
Now that I added headers to relevant tests that need them, I can safely raise InvalidMessageError instead of log and move on.